### PR TITLE
Bug fix for 2.11.11 builds

### PR DESCRIPTION
**Pull request by nh13**
headOption is not found in BetterBufferedIterator for 2.11.11, but is
found for 2.12.2

### CODE REVIEW

File: src/main/scala/com/fulcrumgenomics/commons/collection/BetterBufferedIterator.scala [hunk 1]
```diff
@@ -26,11 +26,17 @@ package com.fulcrumgenomics.commons.collection
 
 import com.fulcrumgenomics.commons.CommonsDef._
 
+/** A little trait to allow BetterBufferedIterator to override the headOption method. The latter method
+  * is not found in [[BetterBufferedIterator]] in 2.11.11, but in 2.12.2! */
+sealed trait HeadOption[A] {
```
Comment (tfenne) [inline src/main/scala/com/fulcrumgenomics/commons/collection/BetterBufferedIterator.scala:31]: I don't think that this trait is needed at all since either clients will know the type to be BetterBufferedIterator and so `headOption` will exist, or it won't know the type beyond "BufferedIterator" in which case it can't call headOption.  I don't think adding the trait gives you anything beyond allowing you to add `override` below.

File: src/main/scala/com/fulcrumgenomics/commons/collection/BetterBufferedIterator.scala [hunk 2]
```diff
@@ -26,11 +26,17 @@ package com.fulcrumgenomics.commons.collection
...
+  def headOption: Option[A]
+}
+
 /**
   * A better buffered iterator that provides implementations of takeWhile and dropWhile
   * that don't discard extra values.
   */
-class BetterBufferedIterator[A](private val iterator: Iterator[A]) extends BufferedIterator[A] {
+class BetterBufferedIterator[A](private val iterator: Iterator[A]) extends HeadOption[A] with scala.collection.BufferedIterator[A] {
```
Comment (tfenne) [inline src/main/scala/com/fulcrumgenomics/commons/collection/BetterBufferedIterator.scala]: Why the change to using the FQN?

**Review by tfenne — COMMENTED** [review]
Couple of minor comments.

**Comment by tfenne** [comment]
Oh .... also maybe a trivial test that uses `headOption` to confirm it works the same under 2.11 and 2.12?

File: src/main/scala/com/fulcrumgenomics/commons/collection/BetterBufferedIterator.scala (hunk 1 above)
Comment (nh13) [reply src/main/scala/com/fulcrumgenomics/commons/collection/BetterBufferedIterator.scala:31]: Since I need to use the `override` keyword when `headOption` is found in `BufferedIterator` in 2.12.

File: src/main/scala/com/fulcrumgenomics/commons/collection/BetterBufferedIterator.scala (hunk 2 above)
Comment (nh13) [reply src/main/scala/com/fulcrumgenomics/commons/collection/BetterBufferedIterator.scala]: Good catch

File: src/main/scala/com/fulcrumgenomics/commons/collection/BetterBufferedIterator.scala [hunk 3]
```diff
@@ -59,6 +65,13 @@ class BetterBufferedIterator[A](private val iterator: Iterator[A]) extends Buffe
   /** Drops items while they match the predicate. */
   override def dropWhile(p: (A) => Boolean): this.type = {
     while (hasNext && p(head)) next()
+    this.headOption
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/commons/collection/BetterBufferedIterator.scala]: removed!
